File: app/src/main/java/ch/ralena/natibo/adapter/LanguageListAdapter.java
package ch.ralena.natibo.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

import ch.ralena.natibo.R;
import ch.ralena.natibo.object.Language;
import io.reactivex.subjects.PublishSubject;

public class LanguageListAdapter extends RecyclerView.Adapter<LanguageListAdapter.ViewHolder> {

	PublishSubject<Language> languageSubject = PublishSubject.create();

	public PublishSubject<Language> asObservable() {
		return languageSubject;
	}

	private ArrayList<Language> languages;

	public LanguageListAdapter(ArrayList<Language> languages) {
		this.languages = languages;
	}

	@NonNull
	@Override
	public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		View view;
		view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_language_list, parent, false);
		return new ViewHolder(view);
	}

	@Override
	public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
		if (position < getItemCount())
			holder.bindView(languages.get(position));
	}

	@Override
	public int getItemCount() {
		return languages.size();
	}

	class ViewHolder extends RecyclerView.ViewHolder {
		private View view;
		private TextView languageName;
		private TextView numPacks;
		private TextView numSentences;
		private ImageView flagImage;
		private Language language;

		ViewHolder(View view) {
			super(view);
			this.view = view;
			languageName = view.findViewById(R.id.languageLabel);
			numPacks = view.findViewById(R.id.numPacksLabel);
			numSentences = view.findViewById(R.id.numSentencesLabel);
			flagImage = view.findViewById(R.id.flagImageView);
			this.view.setOnClickListener(v -> languageSubject.onNext(language));
		}

		void bindView(Language language) {
			this.language = language;
			languageName.setText(language.getLongName());
			numPacks.setText("" + language.getPacks().size());
			numSentences.setText("" + language.getSentenceCount());
			flagImage.setImageResource(language.getLanguageType().getDrawable());
		}
	}
}

File: app/src/main/java/ch/ralena/natibo/ui/language/list/adapter/LanguageListAdapter.kt
package ch.ralena.natibo.ui.language.list.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ch.ralena.natibo.R
import ch.ralena.natibo.data.room.`object`.LanguageRoom
import ch.ralena.natibo.data.room.`object`.LanguageWithPacks
import ch.ralena.natibo.ui.base.BaseRecyclerAdapter
import java.util.*
import javax.inject.Inject

class LanguageListAdapter @Inject constructor(
		private val languages: ArrayList<LanguageWithPacks>
) : BaseRecyclerAdapter<LanguageListAdapter.ViewHolder, LanguageListAdapter.Listener>() {
	interface Listener {
		fun onLanguageClicked(language: LanguageRoom)
	}

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_language_list, parent, false)
		return ViewHolder(view)
	}

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		if (position < itemCount) holder.bindView(languages[position])
	}

	override fun getItemCount(): Int {
		return languages.size
	}

	fun loadLanguages(languages: List<LanguageWithPacks>) {
		this.languages.clear()
		this.languages.addAll(languages)
		notifyDataSetChanged()
	}

	inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
		private val languageName: TextView = view.findViewById(R.id.languageLabel)
		private val numPacks: TextView = view.findViewById(R.id.numPacksLabel)
		private val numSentences: TextView = view.findViewById(R.id.numSentencesLabel)
		private val flagImage: ImageView = view.findViewById(R.id.flag_image)

		private lateinit var languageWithPacks: LanguageWithPacks

		init {
			view.setOnClickListener { for (l in listeners) l.onLanguageClicked(languageWithPacks.language) }
		}

		fun bindView(languageWithPacks: LanguageWithPacks) {
			this.languageWithPacks = languageWithPacks
			val language = languageWithPacks.language
			val packs = languageWithPacks.packs
			languageName.text = language.name
			numPacks.text = "${packs.size}"
//			numSentences.text = "${language.sentenceCount}"
			flagImage.setImageResource(language.flagDrawable)
		}
	}
}